connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectorConfig.java

```diff
@@ -499,13 +516,15 @@ ConfigDef getConfigDefFromConfigProvidingClass(String key, Class<?> cls) {
                         aliasKind + " is abstract and cannot be created. Did you mean " + childClassNames + "?";
                 throw new ConfigException(key, String.valueOf(cls), message);
             }
-            T transformation;
+            T plugin;
             try {
-                transformation = Utils.newInstance(cls, baseClass);
+                plugin = Utils.newInstance(cls, baseClass);
             } catch (Exception e) {
-                throw new ConfigException(key, String.valueOf(cls), "Error getting config definition from " + baseClass.getSimpleName() + ": " + e.getMessage());
+                // Log the entire exception here in order to provide a stack trace that can be useful for debugging classloading issues
+                log.error("Failed to instantiate {} '{}'", baseClass.getSimpleName(), cls, e);
```

Comment:
Added this error message based on discussion of a similar error-handling situation [here](https://github.com/apache/kafka/pull/11349#discussion_r715452885).